packages/api-docs-builder/utils/parseStyles.ts

```diff
@@ -7,87 +9,66 @@ export interface Styles {
   descriptions: Record<string, string>;
 }
 
-export default async function parseStyles(
-  api: { props?: any; filename: string; name: string },
-  program: ttp.ts.Program,
-): Promise<Styles> {
-  // component has no classes or no props
-  // or they're inherited from an external component and we don't want them documented on this component.
-  if (api.props?.classes === undefined) {
-    return {
-      classes: [],
-      descriptions: {},
-      globalClasses: {},
-      name: null,
-    };
-  }
-
-  const typesFilename = api.filename.replace(/\.js$/, '.d.ts');
-  const proptypes = ttp.parseFromProgram(typesFilename, program, {
-    shouldResolveObject: ({ name }) => {
-      return name === 'classes';
-    },
-    checkDeclarations: true,
-  });
+const EMPTY_STYLES: Styles = {
+  classes: [],
+  descriptions: {},
+  globalClasses: {},
+  name: null,
+};
 
-  const component = proptypes.body.find((internalComponent) => {
-    return internalComponent.name === api.name;
-  });
-  if (component === undefined) {
-    return {
-      classes: [],
-      descriptions: {},
-      globalClasses: {},
-      name: null,
-    };
-    // TODO: should we throw?
-    // throw new TypeError(
-    //   `Unable to find declaration of ${api.name} in one of the ${
-    //     proptypes.body.length
-    //   } components: ${proptypes.body.map(({ name }) => name)}`,
-    // );
+function removeUndefinedFromType(type: ts.Type) {
+  // eslint-disable-next-line no-bitwise
+  if (type.flags & ts.TypeFlags.Union) {
+    return (type as ts.UnionType).types.find((subType) => {
+      return subType.flags !== ts.TypeFlags.Undefined;
+    });
   }
 
-  const classes = component.types.find((propType) => {
-    const isClassesProp = propType.name === 'classes';
+  return type;
+}
 
-    return isClassesProp;
-  });
+const getSymbolDescription = (symbol: ts.Symbol, project: TypeScriptProject) =>
+  symbol
+    .getDocumentationComment(project.checker)
+    .flatMap((comment) => comment.text.split('\n'))
+    .filter((line) => !line.startsWith('TODO'))
+    .join('\n');
 
-  let classesPropType: ttp.InterfaceType | undefined;
-  if (classes?.propType.type === 'InterfaceNode') {
-    // classes: {}
-    classesPropType = classes.propType;
-  } else if (classes?.propType.type === 'UnionNode') {
-    // classes?: {}
-    classesPropType = classes.propType.types.find((propType): propType is ttp.InterfaceType => {
-      return propType.type === 'InterfaceNode';
-    });
+export default function parseStyles({
+  project,
+  componentName,
+}: {
+  project: TypeScriptProject;
+  componentName: string;
+}): Styles {
+  const exportedSymbol =
```

Comment:
I need the `Unstable_` version for `Unstable_Grid` (and a lot of picker components).

In the long term, I think it would be better to just loop upon the exported elements instead of looking at the component files.
But that's a way bigger change.